lxd/operations/operations.go

```diff
@@ -110,6 +112,8 @@ type Operation struct {
 	dbOpType    operationtype.Type
 	requestor   *api.EventLifecycleRequestor
 	logger      logger.Logger
+	async       bool
```

Comment:
I'm a bit confused by this. Aren't all operations asynchronous by definition? Can we determine this by the presence of the requestURL field?

Reply:
I was afraid of there being some request that waits for the operation it spawns to end before returning, if there isn't then this field could be discarted and we would just use requestURL as you suggested.

Reply:
Indeed it seems there are such requests, [this](https://github.com/hamistao/lxd/blob/24224b71d05dd049bac9379cb0bb61feeecf225b/lxd/instance_post.go#L740) seems to be one of them

Reply:
link broken

Reply:
How many of those types of operations are there, and might it be more appropriate to change them to be normal synchronous requests, rather than hacking in a concept of a synchronous-asynchronous-operation (which to my mind is super confusing) :)

Reply:
It seems that there is only this one I sent (link is fixed). But thinking about making this more future proof, would it be best to cancel the increment on the completed request metric inside the operation when calling `.Wait()` on that operation?

This way the `async` attribute could be discarted and we wouldn't need to worry about introducing operations that are created and completed inside the same handler. The only downside I see would be the possibility of the operation completing before the `Wait()` call, in which case the metric would be increased anyway, but I believe this should rarely (if ever) happen.